compiler/il/OMRNode.cpp

```diff
@@ -528,6 +540,9 @@ OMR::Node::recreateAndCopyValidPropertiesImpl(TR::Node *originalNode, TR::ILOpCo
    TR_ASSERT(originalNode != NULL, "trying to recreate node from a NULL originalNode.");
    if (originalNode->getOpCodeValue() == op)
       {
+      if (!originalNode->hasSymbolReference() || newSymRef != originalNode->getSymbolReference())
```

Comment:
Again could you please elaborate on why you are making this change ?

Reply:
The rational behind this change and the change above is that `doNotProfile` is used by `supportsInduceOSR` to determine if a node is created in ILGen. Nodes created in ilgen implies that the semantics of the node is equivalent of the semantics of the bytecode. If a node is transmuted, its semantics may have changed and OSR may not be possible.